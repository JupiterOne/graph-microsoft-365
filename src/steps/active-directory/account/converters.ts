import {
  createIntegrationEntity,
  Entity,
  IntegrationInstance,
} from '@jupiterone/integration-sdk-core';
import { Organization } from '@microsoft/microsoft-graph-types';
import { Entities } from '../constants';

export function createAccountEntity(instance: IntegrationInstance): Entity {
  return createIntegrationEntity({
    entityData: {
      source: {},
      assign: {
        _class: Entities.ACCOUNT._class,
        _key: `${Entities.ACCOUNT._type}-${instance.id}`,
        _type: Entities.ACCOUNT._type,
        name: instance.name,
        displayName: instance.name,
      },
    },
  });
}

export function createAccountEntityWithOrganization(
  instance: IntegrationInstance,
  organization: Organization,
  intuneConfig: {
    mobileDeviceManagementAuthority?: string;
    subscriptionState?: string;
    intuneAccountID?: string;
  },
): Entity {
  let defaultDomain: string | undefined;
  const verifiedDomains = organization.verifiedDomains?.map((e) => {
    if (e.isDefault) {
      defaultDomain = e.name as string | undefined;
    }
    return e.name as string;
  });

  return createIntegrationEntity({
    entityData: {
      source: {
        organization,
        intuneConfig,
      },
      assign: {
        _class: Entities.ACCOUNT._class,
        _key: `${Entities.ACCOUNT._type}-${instance.id}`,
        _type: Entities.ACCOUNT._type,
        id: organization.id,
        name: organization.displayName,
        displayName: instance.name,
        organizationName: organization.displayName,
        defaultDomain,
        verifiedDomains,
        intuneAccountId: intuneConfig?.intuneAccountID,
        mobileDeviceManagementAuthority:
          intuneConfig?.mobileDeviceManagementAuthority,
        intuneSubscriptionState: intuneConfig?.subscriptionState,
      },
    },
  });
}
