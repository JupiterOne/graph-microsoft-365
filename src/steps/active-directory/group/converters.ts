import {
  createDirectRelationship,
  createIntegrationEntity,
  Entity,
  parseTimePropertyValue,
  Relationship,
  RelationshipClass,
} from '@jupiterone/integration-sdk-core';
import { Group } from '@microsoft/microsoft-graph-types';
import { Entities, Relationships } from '../constants';

export function createGroupEntity(data: Group): Entity {
  return createIntegrationEntity({
    entityData: {
      source: {}, // removed due to size
      assign: {
        _class: Entities.GROUP._class,
        _type: Entities.GROUP._type,
        _key: data.id!,
        name: data.displayName,
        displayName: data.displayName as string | undefined,
        id: data.id,
        deletedOn: parseTimePropertyValue(data.deletedDateTime),
        classification: data.classification,
        createdOn: parseTimePropertyValue(data.createdDateTime),
        description: data.description ?? undefined,
        email: data.mailEnabled ? data.mail : undefined,
        mail: data.mailEnabled ? data.mail : undefined,
        mailEnabled: data.mailEnabled,
        mailNickname: data.mailNickname,
        renewedOn: parseTimePropertyValue(data.renewedDateTime),
        securityEnabled: data.securityEnabled,
      },
    },
  });
}

export function createAccountGroupRelationship(
  account: Entity,
  group: Entity,
): Relationship {
  return createDirectRelationship({
    _class: RelationshipClass.HAS,
    from: account,
    to: group,
    properties: {
      _type: Relationships.ACCOUNT_HAS_GROUP._type,
    },
  });
}
