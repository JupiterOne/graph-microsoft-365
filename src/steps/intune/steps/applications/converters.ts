import {
  IntegrationEntityData,
  parseTimePropertyValue,
} from '@jupiterone/integration-sdk-core';
import {
  AndroidManagedStoreApp,
  ManagedApp,
  WebApp,
  IosLobApp,
  AndroidLobApp,
  WindowsPhoneXAP,
  MobileLobApp,
  DetectedApp,
} from '@microsoft/microsoft-graph-types-beta';
import { entities } from '../../constants';

// https://docs.microsoft.com/en-us/graph/api/resources/intune-shared-mobileapp?view=graph-rest-beta
export function mapManagedApplicationEntityValues(
  managedApp: ManagedApp & { '@odata.type': string },
): IntegrationEntityData['assign'] {
  return {
    _class: entities.MANAGED_APPLICATION._class,
    _type: entities.MANAGED_APPLICATION._type,
    id: managedApp.id,
    name: managedApp.displayName?.toLowerCase(),
    displayName: managedApp.displayName as string,
    description: managedApp.description,
    notes: managedApp.notes ? [managedApp.notes] : [],
    COTS: !isLineOfBusiness(managedApp['@odata.type']),
    external: !isLineOfBusiness(managedApp['@odata.type']),
    mobile: isMobile(managedApp['@odata.type']),
    productionURL:
      (managedApp as WebApp).appUrl ??
      (managedApp as AndroidManagedStoreApp).appStoreUrl,
    publisher: managedApp.publisher,
    isPublished: managedApp.publishingState === 'published', // Essentially if it is available for download
    createdOn: parseTimePropertyValue(managedApp.createdDateTime),
    lastUpdatedOn: parseTimePropertyValue(managedApp.lastModifiedDateTime),
    featured: managedApp.isFeatured, // Indicates that they are featuring this app on their Company Portal
    privacyInformationURL: managedApp.privacyInformationUrl,
    informationURL: managedApp.informationUrl,
    owner: managedApp.owner || undefined, // Ex: Microsoft, Google, Facebook...
    developer: managedApp.developer, // Almost always the same as the owner

    // Line of Business Apps
    newestVersion: findNewestVersion(managedApp),
    committedContentVersion: (managedApp as MobileLobApp)
      .committedContentVersion,
    packageId: (managedApp as AndroidLobApp).packageId,
  };
}

// https://docs.microsoft.com/en-us/graph/api/resources/intune-devices-detectedapp?view=graph-rest-beta
export function mapDetectedApplicationEntityValues(
  detectedApp: DetectedApp,
): IntegrationEntityData['assign'] {
  return {
    _class: entities.DETECTED_APPLICATION._class,
    _type: entities.DETECTED_APPLICATION._type,
    id: detectedApp.id,
    name: detectedApp.displayName?.toLowerCase(),
    displayName: detectedApp.displayName as string,
    sizeInByte: detectedApp.sizeInByte,
  };
}

export function findNewestVersion(
  managedApp: ManagedApp & WindowsPhoneXAP & AndroidLobApp & IosLobApp,
) {
  return (
    managedApp.identityVersion ??
    managedApp.versionName ??
    managedApp.versionCode ??
    managedApp.versionNumber ??
    managedApp.version ??
    'unversioned'
  );
}

/**
 * Line of business apps need to be manually uploaded to Azure ensuring that they are custom.
 * All other managed apps go throuhg an app store or a website (webApp).
 *
 * @param dataModel Microsoft datatype for the api response.
 * Examples: "#microsoft.graph.webApp", "#microsoft.graph.managedIOSStoreApp", "#microsoft.graph.androidLobApp"
 */
function isLineOfBusiness(dataModel: string) {
  return dataModel.toLowerCase().includes('lob');
}

/**
 *
 * @param dataModel Microsoft datatype for the api response.
 * Examples: "#microsoft.graph.androidStoreApp", "#microsoft.graph.managedIOSStoreApp", "#microsoft.graph.windowsMobileMSI", "#microsoft.graph.officeSuiteApp"
 */
function isMobile(dataModel: string) {
  return [
    'ios', // matches iPhone managed app types
    'android', // matches Android managed app types
    'mobile', // matches Windows phone app types
    'webApp', // webApps are availble for both mobile and desktop
  ].some((el) => dataModel.toLowerCase().indexOf(el) > -1);
}
