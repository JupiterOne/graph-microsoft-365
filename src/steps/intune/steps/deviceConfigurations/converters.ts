import { createIntegrationEntity } from '@jupiterone/integration-sdk-core';
import { DeviceConfiguration } from '@microsoft/microsoft-graph-types-beta';
import { entities } from '../../constants';
import { last } from 'lodash';
import { DeviceConfigurationEntity } from '../../types';

/**
 * There are several different types of device configuration based on your OS and what you are configuring.
 * So many that it does not make sense to ingest each individually until that granularity is desired,
 * or necessary. Because of that, this method ingests all device configurations the same way, and if
 * more granularity is desired, individual configuration settings can be viewed in the raw data.
 *
 * https://docs.microsoft.com/en-us/graph/api/resources/intune-shared-deviceconfiguration?view=graph-rest-beta
 */
export function createDeviceConfigurationEntity(
  deviceConfiguration: DeviceConfiguration,
): DeviceConfigurationEntity {
  return createIntegrationEntity({
    entityData: {
      source: deviceConfiguration,
      assign: {
        _class: entities.DEVICE_CONFIGURATION._class,
        _type: entities.DEVICE_CONFIGURATION._type,
        id: deviceConfiguration.id,
        name: deviceConfiguration.displayName,
        description:
          deviceConfiguration.description || deviceConfiguration.displayName,
        displayName: deviceConfiguration.displayName,
        version: deviceConfiguration.version,
        configurationType: last(deviceConfiguration['@odata.type']?.split('.')), // Possable values are many, but some examples are iosCustomConfiguration, windows10GeneralConfiguration, iosWiFiConfiguration
      },
    },
  });
}
