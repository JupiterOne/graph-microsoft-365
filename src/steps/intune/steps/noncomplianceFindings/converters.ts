import {
  createDirectRelationship,
  createIntegrationEntity,
  parseTimePropertyValue,
  Relationship,
} from '@jupiterone/integration-sdk-core';
import { DeviceConfigurationDeviceStatus } from '@microsoft/microsoft-graph-types-beta';
import { entities, relationships } from '../../constants';
import {
  DeviceConfigurationEntity,
  ManagedDeviceEntity,
  NoncomplianceFindingEntity,
} from '../../types';

// https://docs.microsoft.com/en-us/graph/api/resources/intune-deviceconfig-deviceconfigurationdevicestatus?view=graph-rest-beta
export function createNoncomplianceFindingEntity(
  deviceStatus: DeviceConfigurationDeviceStatus,
  deviceConfigurationEntity: DeviceConfigurationEntity,
): NoncomplianceFindingEntity {
  return createIntegrationEntity({
    entityData: {
      source: deviceStatus,
      assign: {
        _class: entities.NONCOMPLIANCE_FINDING._class,
        _type: entities.NONCOMPLIANCE_FINDING._type,
        id: deviceStatus.id,
        name: 'Latest finding from ' + deviceConfigurationEntity.displayName,
        category: 'endpoint',
        assessment: deviceConfigurationEntity.displayName,
        status: deviceStatus.status, // Possible values are: unknown, notApplicable, compliant, remediated, nonCompliant, error, conflict, notAssigned.
        severity: calculateSeverity(deviceStatus.status),
        numericSeverity: calculateNumericSeverity(deviceStatus.status),
        open: findingIsOpen(deviceStatus.status),
        lastProcessedOn: parseTimePropertyValue(
          deviceStatus.lastReportedDateTime,
        ),
        lastTestedOn: parseTimePropertyValue(deviceStatus.lastReportedDateTime),
        lastUpdatedOn: parseTimePropertyValue(
          deviceStatus.lastReportedDateTime,
        ),
      },
    },
  }) as NoncomplianceFindingEntity;
}

export function findingIsOpen(
  deviceStatus: DeviceConfigurationDeviceStatus['status'],
) {
  return ![
    'notApplicable',
    'compliant',
    'remediated',
    'notAssigned',
    'unknown',
    undefined,
  ].includes(deviceStatus);
}

function calculateNumericSeverity(
  deviceStatus: DeviceConfigurationDeviceStatus['status'],
) {
  if (!findingIsOpen(deviceStatus)) {
    return 1;
  }
  switch (deviceStatus) {
    case 'nonCompliant':
    case 'error':
    case 'conflict':
      return 6;
    default:
      return 1; // This should not happen as we should not create findings without the above statuses
  }
}

function calculateSeverity(
  deviceStatus: DeviceConfigurationDeviceStatus['status'],
) {
  const numbericSeverity = calculateNumericSeverity(deviceStatus);
  if (1 <= numbericSeverity && numbericSeverity <= 2) {
    return 'informational';
  } else if (3 <= numbericSeverity && numbericSeverity <= 4) {
    return 'low';
  } else if (5 <= numbericSeverity && numbericSeverity <= 6) {
    return 'medium';
  } else if (7 <= numbericSeverity && numbericSeverity <= 8) {
    return 'high';
  } else if (9 <= numbericSeverity && numbericSeverity <= 10) {
    return 'critical';
  } else {
    return 'none';
  }
}

export function createDeviceDeviceConfigurationRelationship(
  deviceConfigurationEntity: DeviceConfigurationEntity,
  managedDeviceEntity: ManagedDeviceEntity,
): Relationship {
  return createDirectRelationship({
    _class: relationships.DEVICE_USES_DEVICE_CONFIGURATION._class,
    from: managedDeviceEntity,
    to: deviceConfigurationEntity,
  });
}

export function createNoncomplianceFindingRelationship(
  noncomplianceFindingEntity: NoncomplianceFindingEntity,
  managedDeviceEntity: ManagedDeviceEntity,
): Relationship {
  return createDirectRelationship({
    _class: relationships.DEVICE_HAS_NONCOMPLIANCE_FINDING._class,
    from: managedDeviceEntity,
    to: noncomplianceFindingEntity,
  });
}

export function createDeviceConfigurationNonComplianceFindingRelationship(
  deviceConfigurationEityty: DeviceConfigurationEntity,
  noncomplianceFindingEntity: NoncomplianceFindingEntity,
): Relationship {
  return createDirectRelationship({
    _class:
      relationships.DEVICE_CONFIGURATION_IDENTIFIED_NONCOMPLIANCE_FINDING
        ._class,
    from: deviceConfigurationEityty,
    to: noncomplianceFindingEntity,
  });
}
