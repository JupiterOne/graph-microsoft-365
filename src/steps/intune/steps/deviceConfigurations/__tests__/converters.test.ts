import { createDeviceConfigurationEntity } from '../converters';
import { exampleIosWiFiConfiguration } from './fixtures/exampleIosWiFiConfiguration';
import { exampleIosCustomConfiguration } from './fixtures/exampleIosCustomConfiguration';
import { exampleWindows10GeneralConfiguration } from './fixtures/exampleWindows10GeneralConfiguration';

describe('createManagedDeviceEntity', () => {
  test('iosCustomConfiguration', () => {
    expect(
      createDeviceConfigurationEntity(exampleIosCustomConfiguration),
    ).toMatchSnapshot('iosCustomConfiguration');
  });
  test('iosWifiConfiguration', () => {
    expect(
      createDeviceConfigurationEntity(exampleIosWiFiConfiguration),
    ).toMatchSnapshot('iosWifiConfiguration');
  });
  test('windows10GeneralConfiguration', () => {
    expect(
      createDeviceConfigurationEntity(exampleWindows10GeneralConfiguration),
    ).toMatchSnapshot('windows10GeneralConfiguration');
  });
});
