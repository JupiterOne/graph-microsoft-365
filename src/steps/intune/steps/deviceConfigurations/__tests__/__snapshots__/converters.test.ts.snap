// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createManagedDeviceEntity iosCustomConfiguration: iosCustomConfiguration 1`] = `
Object {
  "_class": Array [
    "Configuration",
  ],
  "_key": "5b748b7c-8b6a-41fd-bd97-f509fdf5caf5",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "@odata.type": "#microsoft.graph.iosWiFiConfiguration",
        "connectAutomatically": false,
        "connectWhenNetworkNameIsHidden": false,
        "createdDateTime": "2021-02-25T21:47:36.8239034Z",
        "description": "Test iOS wifi policy",
        "deviceManagementApplicabilityRuleDeviceMode": null,
        "deviceManagementApplicabilityRuleOsEdition": null,
        "deviceManagementApplicabilityRuleOsVersion": null,
        "disableMacAddressRandomization": null,
        "displayName": "Test iOS Wifi policy",
        "id": "5b748b7c-8b6a-41fd-bd97-f509fdf5caf5",
        "lastModifiedDateTime": "2021-02-25T21:47:36.8239034Z",
        "networkName": "fake network",
        "preSharedKey": null,
        "proxyAutomaticConfigurationUrl": null,
        "proxyManualAddress": null,
        "proxyManualPort": null,
        "proxySettings": "none",
        "roleScopeTagIds": Array [
          "0",
        ],
        "ssid": "FakeyFakerson",
        "supportsScopeTags": true,
        "version": 1,
        "wiFiSecurityType": "open",
      },
    },
  ],
  "_type": "intune_device_configuration",
  "configurationType": "iosWiFiConfiguration",
  "createdOn": 1614289656823,
  "description": "Test iOS wifi policy",
  "displayName": "Test iOS Wifi policy",
  "id": "5b748b7c-8b6a-41fd-bd97-f509fdf5caf5",
  "name": "Test iOS Wifi policy",
  "version": 1,
}
`;

exports[`createManagedDeviceEntity iosWifiConfiguration: iosWifiConfiguration 1`] = `
Object {
  "_class": Array [
    "Configuration",
  ],
  "_key": "5b748b7c-8b6a-41fd-bd97-f509fdf5caf5",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "@odata.type": "#microsoft.graph.iosWiFiConfiguration",
        "connectAutomatically": false,
        "connectWhenNetworkNameIsHidden": false,
        "createdDateTime": "2021-02-25T21:47:36.8239034Z",
        "description": "Test iOS wifi policy",
        "deviceManagementApplicabilityRuleDeviceMode": null,
        "deviceManagementApplicabilityRuleOsEdition": null,
        "deviceManagementApplicabilityRuleOsVersion": null,
        "disableMacAddressRandomization": null,
        "displayName": "Test iOS Wifi policy",
        "id": "5b748b7c-8b6a-41fd-bd97-f509fdf5caf5",
        "lastModifiedDateTime": "2021-02-25T21:47:36.8239034Z",
        "networkName": "fake network",
        "preSharedKey": null,
        "proxyAutomaticConfigurationUrl": null,
        "proxyManualAddress": null,
        "proxyManualPort": null,
        "proxySettings": "none",
        "roleScopeTagIds": Array [
          "0",
        ],
        "ssid": "FakeyFakerson",
        "supportsScopeTags": true,
        "version": 1,
        "wiFiSecurityType": "open",
      },
    },
  ],
  "_type": "intune_device_configuration",
  "configurationType": "iosWiFiConfiguration",
  "createdOn": 1614289656823,
  "description": "Test iOS wifi policy",
  "displayName": "Test iOS Wifi policy",
  "id": "5b748b7c-8b6a-41fd-bd97-f509fdf5caf5",
  "name": "Test iOS Wifi policy",
  "version": 1,
}
`;

exports[`createManagedDeviceEntity windows10GeneralConfiguration: windows10GeneralConfiguration 1`] = `
Object {
  "_class": Array [
    "Configuration",
  ],
  "_key": "c271a86c-a62a-4655-ac1e-a28f67fd3e18",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "@odata.type": "#microsoft.graph.windows10GeneralConfiguration",
        "accountsBlockAddingNonMicrosoftAccountEmail": false,
        "activateAppsWithVoice": "notConfigured",
        "antiTheftModeBlocked": false,
        "appManagementMSIAllowUserControlOverInstall": false,
        "appManagementMSIAlwaysInstallWithElevatedPrivileges": false,
        "appManagementPackageFamilyNamesToLaunchAfterLogOn": Array [],
        "appsAllowTrustedAppsSideloading": "notConfigured",
        "appsBlockWindowsStoreOriginatedApps": false,
        "authenticationAllowSecondaryDevice": false,
        "authenticationPreferredAzureADTenantDomainName": null,
        "authenticationWebSignIn": "notConfigured",
        "bluetoothAllowedServices": Array [],
        "bluetoothBlockAdvertising": false,
        "bluetoothBlockDiscoverableMode": false,
        "bluetoothBlockPrePairing": false,
        "bluetoothBlockPromptedProximalConnections": false,
        "bluetoothBlocked": false,
        "cameraBlocked": false,
        "cellularBlockDataWhenRoaming": false,
        "cellularBlockVpn": false,
        "cellularBlockVpnWhenRoaming": false,
        "cellularData": "allowed",
        "certificatesBlockManualRootCertificateInstallation": false,
        "configureTimeZone": null,
        "connectedDevicesServiceBlocked": false,
        "copyPasteBlocked": false,
        "cortanaBlocked": false,
        "createdDateTime": "2021-02-25T20:56:29.1537954Z",
        "cryptographyAllowFipsAlgorithmPolicy": false,
        "dataProtectionBlockDirectMemoryAccess": false,
        "defenderBlockEndUserAccess": false,
        "defenderBlockOnAccessProtection": false,
        "defenderCloudBlockLevel": "notConfigured",
        "defenderCloudExtendedTimeout": null,
        "defenderCloudExtendedTimeoutInSeconds": null,
        "defenderDaysBeforeDeletingQuarantinedMalware": null,
        "defenderDetectedMalwareActions": null,
        "defenderDisableCatchupFullScan": false,
        "defenderDisableCatchupQuickScan": false,
        "defenderFileExtensionsToExclude": Array [],
        "defenderFilesAndFoldersToExclude": Array [],
        "defenderMonitorFileActivity": "userDefined",
        "defenderPotentiallyUnwantedAppAction": null,
        "defenderPotentiallyUnwantedAppActionSetting": "userDefined",
        "defenderProcessesToExclude": Array [],
        "defenderPromptForSampleSubmission": "userDefined",
        "defenderRequireBehaviorMonitoring": false,
        "defenderRequireCloudProtection": false,
        "defenderRequireNetworkInspectionSystem": false,
        "defenderRequireRealTimeMonitoring": false,
        "defenderScanArchiveFiles": false,
        "defenderScanDownloads": false,
        "defenderScanIncomingMail": false,
        "defenderScanMappedNetworkDrivesDuringFullScan": false,
        "defenderScanMaxCpu": null,
        "defenderScanNetworkFiles": false,
        "defenderScanRemovableDrivesDuringFullScan": false,
        "defenderScanScriptsLoadedInInternetExplorer": false,
        "defenderScanType": "userDefined",
        "defenderScheduleScanEnableLowCpuPriority": false,
        "defenderScheduledQuickScanTime": null,
        "defenderScheduledScanTime": null,
        "defenderSignatureUpdateIntervalInHours": null,
        "defenderSubmitSamplesConsentType": "sendSafeSamplesAutomatically",
        "defenderSystemScanSchedule": "userDefined",
        "description": "Description",
        "developerUnlockSetting": "notConfigured",
        "deviceManagementApplicabilityRuleDeviceMode": null,
        "deviceManagementApplicabilityRuleOsEdition": Object {
          "name": null,
          "osEditionTypes": Array [
            "windows10Education",
          ],
          "ruleType": "include",
        },
        "deviceManagementApplicabilityRuleOsVersion": null,
        "deviceManagementBlockFactoryResetOnMobile": false,
        "deviceManagementBlockManualUnenroll": false,
        "diagnosticsDataSubmissionMode": "userDefined",
        "displayAppListWithGdiDPIScalingTurnedOff": Array [],
        "displayAppListWithGdiDPIScalingTurnedOn": Array [],
        "displayName": "Test Windows 10 Device Restrictions Configuration Profile",
        "edgeAllowStartPagesModification": false,
        "edgeBlockAccessToAboutFlags": false,
        "edgeBlockAddressBarDropdown": false,
        "edgeBlockAutofill": false,
        "edgeBlockCompatibilityList": false,
        "edgeBlockDeveloperTools": false,
        "edgeBlockEditFavorites": false,
        "edgeBlockExtensions": false,
        "edgeBlockFullScreenMode": false,
        "edgeBlockInPrivateBrowsing": false,
        "edgeBlockJavaScript": false,
        "edgeBlockLiveTileDataCollection": false,
        "edgeBlockPasswordManager": false,
        "edgeBlockPopups": false,
        "edgeBlockPrelaunch": false,
        "edgeBlockPrinting": false,
        "edgeBlockSavingHistory": false,
        "edgeBlockSearchEngineCustomization": false,
        "edgeBlockSearchSuggestions": false,
        "edgeBlockSendingDoNotTrackHeader": false,
        "edgeBlockSendingIntranetTrafficToInternetExplorer": false,
        "edgeBlockSideloadingExtensions": false,
        "edgeBlockTabPreloading": false,
        "edgeBlockWebContentOnNewTabPage": false,
        "edgeBlocked": false,
        "edgeClearBrowsingDataOnExit": false,
        "edgeCookiePolicy": "userDefined",
        "edgeDisableFirstRunPage": false,
        "edgeEnterpriseModeSiteListLocation": null,
        "edgeFavoritesBarVisibility": "notConfigured",
        "edgeFavoritesListLocation": null,
        "edgeFirstRunUrl": null,
        "edgeHomeButtonConfiguration": null,
        "edgeHomeButtonConfigurationEnabled": false,
        "edgeHomepageUrls": Array [],
        "edgeKioskModeRestriction": "notConfigured",
        "edgeKioskResetAfterIdleTimeInMinutes": null,
        "edgeNewTabPageURL": null,
        "edgeOpensWith": "notConfigured",
        "edgePreventCertificateErrorOverride": false,
        "edgeRequireSmartScreen": false,
        "edgeRequiredExtensionPackageFamilyNames": Array [],
        "edgeSearchEngine": null,
        "edgeSendIntranetTrafficToInternetExplorer": false,
        "edgeShowMessageWhenOpeningInternetExplorerSites": "notConfigured",
        "edgeSyncFavoritesWithInternetExplorer": false,
        "edgeTelemetryForMicrosoft365Analytics": "notConfigured",
        "enableAutomaticRedeployment": false,
        "energySaverOnBatteryThresholdPercentage": null,
        "energySaverPluggedInThresholdPercentage": null,
        "enterpriseCloudPrintDiscoveryEndPoint": null,
        "enterpriseCloudPrintDiscoveryMaxLimit": null,
        "enterpriseCloudPrintMopriaDiscoveryResourceIdentifier": null,
        "enterpriseCloudPrintOAuthAuthority": null,
        "enterpriseCloudPrintOAuthClientIdentifier": null,
        "enterpriseCloudPrintResourceIdentifier": null,
        "experienceBlockDeviceDiscovery": false,
        "experienceBlockErrorDialogWhenNoSIM": false,
        "experienceBlockTaskSwitcher": false,
        "experienceDoNotSyncBrowserSettings": "notConfigured",
        "findMyFiles": "notConfigured",
        "gameDvrBlocked": false,
        "id": "c271a86c-a62a-4655-ac1e-a28f67fd3e18",
        "inkWorkspaceAccess": "notConfigured",
        "inkWorkspaceAccessState": "notConfigured",
        "inkWorkspaceBlockSuggestedApps": false,
        "internetSharingBlocked": false,
        "lastModifiedDateTime": "2021-02-25T20:56:29.5465785Z",
        "locationServicesBlocked": false,
        "lockScreenActivateAppsWithVoice": "notConfigured",
        "lockScreenAllowTimeoutConfiguration": false,
        "lockScreenBlockActionCenterNotifications": false,
        "lockScreenBlockCortana": false,
        "lockScreenBlockToastNotifications": false,
        "lockScreenTimeoutInSeconds": null,
        "logonBlockFastUserSwitching": false,
        "messagingBlockMMS": false,
        "messagingBlockRichCommunicationServices": false,
        "messagingBlockSync": false,
        "microsoftAccountBlockSettingsSync": false,
        "microsoftAccountBlocked": false,
        "microsoftAccountSignInAssistantSettings": "notConfigured",
        "networkProxyApplySettingsDeviceWide": false,
        "networkProxyAutomaticConfigurationUrl": null,
        "networkProxyDisableAutoDetect": false,
        "networkProxyServer": null,
        "nfcBlocked": false,
        "oneDriveDisableFileSync": false,
        "passwordBlockSimple": false,
        "passwordExpirationDays": null,
        "passwordMinimumAgeInDays": null,
        "passwordMinimumCharacterSetCount": null,
        "passwordMinimumLength": null,
        "passwordMinutesOfInactivityBeforeScreenTimeout": null,
        "passwordPreviousPasswordBlockCount": null,
        "passwordRequireWhenResumeFromIdleState": false,
        "passwordRequired": false,
        "passwordRequiredType": "deviceDefault",
        "passwordSignInFailureCountBeforeFactoryReset": null,
        "personalizationDesktopImageUrl": null,
        "personalizationLockScreenImageUrl": null,
        "powerButtonActionOnBattery": "notConfigured",
        "powerButtonActionPluggedIn": "notConfigured",
        "powerHybridSleepOnBattery": "notConfigured",
        "powerHybridSleepPluggedIn": "notConfigured",
        "powerLidCloseActionOnBattery": "notConfigured",
        "powerLidCloseActionPluggedIn": "notConfigured",
        "powerSleepButtonActionOnBattery": "notConfigured",
        "powerSleepButtonActionPluggedIn": "notConfigured",
        "printerBlockAddition": false,
        "printerDefaultName": null,
        "printerNames": Array [],
        "privacyAdvertisingId": "notConfigured",
        "privacyAutoAcceptPairingAndConsentPrompts": false,
        "privacyBlockActivityFeed": false,
        "privacyBlockInputPersonalization": false,
        "privacyBlockPublishUserActivities": false,
        "privacyDisableLaunchExperience": false,
        "resetProtectionModeBlocked": false,
        "roleScopeTagIds": Array [
          "0",
        ],
        "safeSearchFilter": "userDefined",
        "screenCaptureBlocked": true,
        "searchBlockDiacritics": false,
        "searchBlockWebResults": false,
        "searchDisableAutoLanguageDetection": false,
        "searchDisableIndexerBackoff": false,
        "searchDisableIndexingEncryptedItems": false,
        "searchDisableIndexingRemovableDrive": false,
        "searchDisableLocation": false,
        "searchDisableUseLocation": false,
        "searchEnableAutomaticIndexSizeManangement": false,
        "searchEnableRemoteQueries": false,
        "securityBlockAzureADJoinedDevicesAutoEncryption": false,
        "settingsBlockAccountsPage": false,
        "settingsBlockAddProvisioningPackage": false,
        "settingsBlockAppsPage": false,
        "settingsBlockChangeLanguage": false,
        "settingsBlockChangePowerSleep": false,
        "settingsBlockChangeRegion": false,
        "settingsBlockChangeSystemTime": false,
        "settingsBlockDevicesPage": false,
        "settingsBlockEaseOfAccessPage": false,
        "settingsBlockEditDeviceName": false,
        "settingsBlockGamingPage": false,
        "settingsBlockNetworkInternetPage": false,
        "settingsBlockPersonalizationPage": false,
        "settingsBlockPrivacyPage": false,
        "settingsBlockRemoveProvisioningPackage": false,
        "settingsBlockSettingsApp": false,
        "settingsBlockSystemPage": false,
        "settingsBlockTimeLanguagePage": false,
        "settingsBlockUpdateSecurityPage": false,
        "sharedUserAppDataAllowed": false,
        "smartScreenAppInstallControl": "notConfigured",
        "smartScreenBlockPromptOverride": false,
        "smartScreenBlockPromptOverrideForFiles": false,
        "smartScreenEnableAppInstallControl": false,
        "startBlockUnpinningAppsFromTaskbar": false,
        "startMenuAppListVisibility": "userDefined",
        "startMenuHideChangeAccountSettings": false,
        "startMenuHideFrequentlyUsedApps": false,
        "startMenuHideHibernate": false,
        "startMenuHideLock": false,
        "startMenuHidePowerButton": false,
        "startMenuHideRecentJumpLists": false,
        "startMenuHideRecentlyAddedApps": false,
        "startMenuHideRestartOptions": false,
        "startMenuHideShutDown": false,
        "startMenuHideSignOut": false,
        "startMenuHideSleep": false,
        "startMenuHideSwitchAccount": false,
        "startMenuHideUserTile": false,
        "startMenuLayoutEdgeAssetsXml": null,
        "startMenuLayoutXml": null,
        "startMenuMode": "userDefined",
        "startMenuPinnedFolderDocuments": "notConfigured",
        "startMenuPinnedFolderDownloads": "notConfigured",
        "startMenuPinnedFolderFileExplorer": "notConfigured",
        "startMenuPinnedFolderHomeGroup": "notConfigured",
        "startMenuPinnedFolderMusic": "notConfigured",
        "startMenuPinnedFolderNetwork": "notConfigured",
        "startMenuPinnedFolderPersonalFolder": "notConfigured",
        "startMenuPinnedFolderPictures": "notConfigured",
        "startMenuPinnedFolderSettings": "notConfigured",
        "startMenuPinnedFolderVideos": "notConfigured",
        "storageBlockRemovableStorage": false,
        "storageRequireMobileDeviceEncryption": false,
        "storageRestrictAppDataToSystemVolume": false,
        "storageRestrictAppInstallToSystemVolume": false,
        "supportsScopeTags": true,
        "systemTelemetryProxyServer": null,
        "taskManagerBlockEndTask": false,
        "tenantLockdownRequireNetworkDuringOutOfBoxExperience": false,
        "uninstallBuiltInApps": false,
        "usbBlocked": false,
        "version": 2,
        "voiceRecordingBlocked": false,
        "webRtcBlockLocalhostIpAddress": false,
        "wiFiBlockAutomaticConnectHotspots": false,
        "wiFiBlockManualConfiguration": false,
        "wiFiBlocked": false,
        "wiFiScanInterval": null,
        "windows10AppsForceUpdateSchedule": null,
        "windowsSpotlightBlockConsumerSpecificFeatures": false,
        "windowsSpotlightBlockOnActionCenter": false,
        "windowsSpotlightBlockTailoredExperiences": false,
        "windowsSpotlightBlockThirdPartyNotifications": false,
        "windowsSpotlightBlockWelcomeExperience": false,
        "windowsSpotlightBlockWindowsTips": false,
        "windowsSpotlightBlocked": false,
        "windowsSpotlightConfigureOnLockScreen": "notConfigured",
        "windowsStoreBlockAutoUpdate": false,
        "windowsStoreBlocked": false,
        "windowsStoreEnablePrivateStoreOnly": false,
        "wirelessDisplayBlockProjectionToThisDevice": false,
        "wirelessDisplayBlockUserInputFromReceiver": false,
        "wirelessDisplayRequirePinForPairing": false,
      },
    },
  ],
  "_type": "intune_device_configuration",
  "configurationType": "windows10GeneralConfiguration",
  "createdOn": 1614286589153,
  "description": "Description",
  "displayName": "Test Windows 10 Device Restrictions Configuration Profile",
  "id": "c271a86c-a62a-4655-ac1e-a28f67fd3e18",
  "name": "Test Windows 10 Device Restrictions Configuration Profile",
  "version": 2,
}
`;
